diff --git a/node_modules/@jofr/capacitor-media-session/android/src/main/java/io/github/jofr/capacitor/mediasessionplugin/MediaSessionPlugin.java b/node_modules/@jofr/capacitor-media-session/android/src/main/java/io/github/jofr/capacitor/mediasessionplugin/MediaSessionPlugin.java
index 6f54b7f..3b97346 100644
--- a/node_modules/@jofr/capacitor-media-session/android/src/main/java/io/github/jofr/capacitor/mediasessionplugin/MediaSessionPlugin.java
+++ b/node_modules/@jofr/capacitor-media-session/android/src/main/java/io/github/jofr/capacitor/mediasessionplugin/MediaSessionPlugin.java
@@ -82,17 +82,26 @@ public class MediaSessionPlugin extends Plugin {
     }
 
     public void startMediaService() {
-        Intent intent = new Intent(getActivity(), MediaSessionService.class);
-        ContextCompat.startForegroundService(getContext(), intent);
-        getContext().bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
+        try {
+            Intent intent = new Intent(getActivity(), MediaSessionService.class);
+            ContextCompat.startForegroundService(getContext(), intent);
+            getContext().bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
+        } catch (Exception e) {
+            Log.e(TAG, "Error starting MediaSessionService: " + e.getMessage());
+        }
     }
 
     private void updateServiceMetadata() {
-        service.setTitle(title);
-        service.setArtist(artist);
-        service.setAlbum(album);
-        service.setArtwork(artwork);
-        service.update();
+        try {
+            service.setTitle(title);
+            service.setArtist(artist);
+            service.setAlbum(album);
+            service.setArtwork(artwork);
+            service.update();
+        }
+        catch(Exception e) {
+            Log.e(TAG, "Error updating metadata: " + e.getMessage());
+        }
     }
 
 
@@ -141,6 +150,7 @@ public class MediaSessionPlugin extends Plugin {
     }
 
     private void updateServicePlaybackState() {
+        try {
         if (playbackState.equals("playing")) {
             service.setPlaybackState(PlaybackStateCompat.STATE_PLAYING);
             service.update();
@@ -151,6 +161,9 @@ public class MediaSessionPlugin extends Plugin {
             service.setPlaybackState(PlaybackStateCompat.STATE_NONE);
             service.update();
         }
+        } catch (Exception e) {
+            Log.e(TAG, "Error updating playback state: " + e.getMessage());
+        }
     }
 
     @PluginMethod
@@ -170,11 +183,15 @@ public class MediaSessionPlugin extends Plugin {
     }
 
     private void updateServicePositionState() {
-        service.setDuration(Math.round(duration * 1000));
-        service.setPosition(Math.round(position * 1000));
-        float playbackSpeed = playbackRate == 0.0 ? (float) 1.0 : (float) playbackRate;
-        service.setPlaybackSpeed(playbackSpeed);
-        service.update();
+        try {
+            service.setDuration(Math.round(duration * 1000));
+            service.setPosition(Math.round(position * 1000));
+            float playbackSpeed = playbackRate == 0.0 ? (float) 1.0 : (float) playbackRate;
+            service.setPlaybackSpeed(playbackSpeed);
+            service.update();
+        } catch (Exception e) {
+            Log.e(TAG, "Error updating position state: " + e.getMessage());
+        }
     }
 
     @PluginMethod
